prediction/src/algorithms/segment/trained_model.py

```diff
@@ -36,20 +38,45 @@ def predict(dicom_path, centroids):
         dict: Dictionary containing path to serialized binary masks and
             volumes per centroid with form::
             {'binary_mask_path': str,
-             'volumes': list[float]}
+             'volumes': list[float]
+             'diameters': list[float]}
     """
-    load_ct(dicom_path)
-    segment_path = os.path.join(os.path.dirname(__file__),
-                                'assets', 'test_mask.npy')
-    volumes = calculate_volume(segment_path, centroids)
+
+    reader = sitk.ImageSeriesReader()
+    reader.SetFileNames(reader.GetGDCMSeriesFileNames(dicom_path))
+    image = reader.Execute()
+
+    binary_mask = np.zeros_like(sitk.GetArrayFromImage(image), dtype=bool)
+
+    volumes = []
+    diameters = []
+    for nodule in centroids:
+        mask, diameter, volume = levelset_segment.segment_nodule(image, nodule)
+        volumes.append(volume)
+        diameters.append(diameter)
+        binary_mask |= sitk.GetArrayFromImage(mask).astype(bool)
+
+    segment_path = tempfile.NamedTemporaryFile(suffix='.npy')
+    _FILES.append(segment_path)
```

Comment:
Sorry to mess you around here, but once I saw this implementation I just can't quite help feeling it doesn't feel right. :(

I'm not really sure about the best solution. Can't we just pass around the ``TemporaryFile`` object? Or change the API such that people are passing it *into* this method? At the very very least we should factor our the ``_FILES`` mangling into a global utility (eg. ``get_named_temporary_file``) that isn't local to this module.

Other suggestions highly highly welcome!

Reply:
The problem is that our output is just the path to the mask - we have no way of knowing when the client is done with the file. 
I think it makes sense to move this into a global utility instead.

Reply:
Just to throw out another idea, how about having a `segment_path` kwarg? something like:

```python
def predict(*args, segment_path='default_path.npy'):
    pass
```

Reply:
@lamby Where would we want such a util to live? prediction/src/utils.py?

@reubano I think we still have the fundamental problem that either the client needs to clean up, or the prediction service does?

Reply:
@dchansen Sounds reasonable, does that "feel" good when you look at the code? Could probably live one level up if we have shared utils...?

Reply:
Ok. Moved the functionality into utils. I think this is a reasonable solution?

Reply:
Looks good... but you still need the exit handler right? FYI, it also works as a decorator. From the docs:

```python
import atexit

@atexit.register
def goodbye():
    print("You are now leaving the Python sector.")
```

Reply:
No, that was a mistake on my part. All the files are closed (and thus deleted) when the program exits, so this should be ok.

Reply:
> No, that was a mistake on my part. All the files are closed (and thus deleted) when the program exits, so this should be ok.

Can you add that to the docstring to the util you created?

Also, what's the point of `_FILES` if we aren't cleaning up?